jsettlers.logic/src/main/java/jsettlers/input/GuiInterface.java

```diff
@@ -621,8 +632,17 @@ private void setSelection(SelectionSet selection) {
 
 	@Override
 	public void refreshSelection() {
-		connector.setSelection(null);
-		connector.setSelection(currentSelection);
+		if (!currentSelection.isEmpty()) {
+			SelectionSet newSelection = new SelectionSet();
+
+			for (ISelectable selected : currentSelection) {
+				if (selected.isSelected()) {
+					newSelection.add(selected);
+				}
+			}
+
+			setSelection(newSelection);
```

Comment:
This is a task for the game logic - not the UI.

You have race conditions in this code.